src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -189,33 +137,109 @@ impl Renderer for HtmlHandlebars {
+        Ok(())
+    }
 
-            let name = match custom_file.strip_prefix(book.get_root()) {
-                Ok(p) => p.to_str().expect("Could not convert to str"),
-                Err(_) => {
-                    custom_file
-                        .file_name()
-                        .expect("File has a file name")
-                        .to_str()
-                        .expect("Could not convert to str")
-                }
-            };
+    fn write_custom_file(&self, custom_file: &Path, book: &MDBook) -> Result<(), Box<Error>> {
+        let mut data = Vec::new();
+        let mut f = File::open(custom_file)?;
+        f.read_to_end(&mut data)?;
 
-            book.write_file(name, &data)?;
+        let name = match custom_file.strip_prefix(book.get_root()) {
+            Ok(p) => p.to_str().expect("Could not convert to str"),
+            Err(_) => {
+                custom_file.file_name()
+                    .expect("File has a file name")
+                    .to_str()
+                    .expect("Could not convert to str")
+            },
+        };
+
+        book.write_file(name, &data)?;
+
+        Ok(())
+    }
+
+    /// Update the context with data for this file
+    fn configure_print_version(&self, data: &mut serde_json::Map<String, serde_json::Value>, print_content: &str) {
+        data.insert("path".to_owned(), json!("print.md"));
+        data.insert("content".to_owned(), json!(print_content));
+        data.insert("path_to_root".to_owned(), json!(utils::fs::path_to_root(Path::new("print.md"))));
+    }
+
+    fn register_hbs_helpers(&self, handlebars: &mut Handlebars) {
+        handlebars.register_helper("toc", Box::new(helpers::toc::RenderToc));
+        handlebars.register_helper("previous", Box::new(helpers::navigation::previous));
+        handlebars.register_helper("next", Box::new(helpers::navigation::next));
+    }
+
+    /// Copy across any additional CSS and JavaScript files which the book
+    /// has been configured to use.
+    fn copy_additional_css_and_js(&self, book: &MDBook) -> Result<(), Box<Error>> {
+        let custom_files = book.get_additional_css()
+            .iter()
+            .chain(book.get_additional_js().iter());
+
+        for custom_file in custom_files {
+            self.write_custom_file(custom_file, book)?;
+        }
+
+        Ok(())
+    }
+}
+
+
+impl Renderer for HtmlHandlebars {
+    fn render(&self, book: &MDBook) -> Result<(), Box<Error>> {
+        debug!("[fn]: render");
+        let mut handlebars = Handlebars::new();
+
+        let theme = theme::Theme::new(book.get_theme_path());
+
+        debug!("[*]: Register handlebars template");
+        handlebars.register_template_string("index", String::from_utf8(theme.index.clone())?)?;
+
+        debug!("[*]: Register handlebars helpers");
+        self.register_hbs_helpers(&mut handlebars);
+
+        let mut data = make_data(book)?;
+
+        // Print version
+        let mut print_content = String::new();
```

Comment:
I believe this is for when you'll try to print the html page from your browser, so it's actually a subsection of the HTML renderer. That was my interpretation anyway, either way at the minimum I should pull all print-specific stuff out into its own method.